natlib/svcd.c

```diff
@@ -185,3 +188,82 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+
+//////////////////////////////////////////////////////////////////////////////
+// SVCD.subdispatch() implementation
+// Maintainer: Running with scissors
+//////////////////////////////////////////////////////////////////////////////
+
+static int svcd_subdispatch(lua_State *L) {
+  //local parr = storm.array.fromstr(pay)
+  size_t len;
+  char *parr = lua_tolstring(L, 1, &len);
+  char* srcip = lua_tostring(L, 2); //? what type is the ip?
+  int strcport = lua_tointeger(L, 3);
+
+  //local cmd = parr:get(1);
+  char cmd = parr[0];
+  //local svc_id = parr:get_as(storm.array.UINT16,1);
+  int16_t svc_id = *(int16_t*)(parr+1);
```

Comment:
You cannot do an unaligned access like this on ARM. See my other ranty comments for people doing unaligned writes.